src/BagReader.js

```diff
@@ -242,30 +245,37 @@ export default class BagReader {
       : this._file.size() - chunkInfo.chunkPosition;
 
     this._file.read(chunkInfo.chunkPosition, readLength, (err: Error | null, buffer?: Buffer) => {
-      if (err || !buffer) {
-        return callback(err || new Error("Missing both error and buffer"));
-      }
-
-      const chunk = this.readRecordFromBuffer(buffer, chunkInfo.chunkPosition, Chunk);
-      const { compression } = chunk;
-      if (compression !== "none") {
-        const decompressFn = decompress[compression];
-        if (!decompressFn) {
-          return callback(new Error(`Unsupported compression type ${chunk.compression}`));
+      (async () => {
```

Comment:
Any preferred approach to awaiting the decompression function? Should the `read` function change to support an async callback?